client/lib/interval/runner.js

```diff
@@ -0,0 +1,97 @@
+import { fromJS } from 'immutable';
+
+export const EVERY_SECOND = 1000;
+export const EVERY_FIVE_SECONDS = 5 * 1000;
+export const EVERY_TEN_SECONDS = 10 * 1000;
+export const EVERY_THIRTY_SECONDS = 30 * 1000;
+export const EVERY_MINUTE = 60 * 1000;
+
+const initialState = fromJS( {
+	nextId: 1,
+	periodTimers: {
+		EVERY_SECOND: null,
+		EVERY_FIVE_SECONDS: null,
+		EVERY_TEN_SECONDS: null,
+		EVERY_THIRTY_SECONDS: null,
+		EVERY_MINUTE: null
+	},
+	actions: []
+} );
+let state = initialState;
+
+const increment = a => a + 1;
+const push = item => list => list.push( item );
+const pull = id => list => list.filterNot( o => o.get( 'id' ) === id );
```

Comment:
@rralian I added some documentation and renamed some functions in f200621 - not sure if it will help. Mostly in these lines I'm taking advantage of the fat-arrow function syntax. Would it be clearer to you if it were more verbose?

```js
function push( item ) {
	return function( list ) {
		list.push( item );
	};
}
```